cmd/commands/commands.go

```diff
@@ -270,7 +270,7 @@ func actionDecorator(f func(*cli.Context) error) func(*cli.Context) error {
 		// 'unlock') but the wallet is already unlocked, then these
 		// methods aren't recognized any more because this service is
 		// shut down after successful unlock.
-		if s.Code() == codes.Unknown &&
+		if s.Code() == codes.Unimplemented &&
```

Comment:
I think we need to check for both `Unimplemented` and `Unknown`?

Reply:
No, `Unknown` is the default code that we return everywhere (basically if you don't use `status.Error()` and just return an error, the `Unknown` status is always used). Only `Unimplemented` tells us that we're trying to call into a method that doesn't exist.
We also don't get an error saying "wallet locked" back. If only the wallet unlocker is active, any other gRPC method will return `Unimplemented`.
So unfortunately we can't really tell the difference between the wallet being locked and a gRPC method that's not enabled because of build tags. That's why I updated the error message instead.